roles/metrics-export.lua

```diff
@@ -1,17 +1,274 @@
-local function validate()
+local urilib = require("uri")
+local http_server = require('http.server')
 
+local M = {}
+
+local function is_array(tbl)
+    assert(type(tbl) == "table", "a table expected")
+    for k, _ in pairs(tbl) do
+        local found = false
+        for idx, _ in ipairs(tbl) do
+            if type(k) == type(idx) and k == idx then
+                found = true
+            end
+        end
+        if not found then
+            return false
+        end
+    end
+    return true
+end
+
+-- Removes extra '/' from start and end of the path to avoid paths duplication.
+local function remove_side_slashes(path)
+    if path:startswith('/') then
+        path = string.lstrip(path, '/')
```

Comment:
It's safe to call lstrip/rstrip if there are no symbols, but you may lease these conditions here for readability.

Reply:
> but you may lease these conditions here for readability

Thank you, it was the idea.